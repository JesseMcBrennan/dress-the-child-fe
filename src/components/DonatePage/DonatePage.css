.donate-title {
  font-size: 4rem;
  font-weight: 300; }
