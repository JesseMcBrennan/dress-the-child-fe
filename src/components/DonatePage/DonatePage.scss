.donate-btn {
  color: #d9318a;
  padding: 0.5rem 1.5rem;
  border: 0.1rem solid #d9318a;
  font-size: 2rem;
  text-decoration: none;
  border-radius: 1rem;
  cursor: pointer;
}

.donate-btn-hidden {
  display: none;
}

.donate-btn:hover {
  color: #3574b6;
  border: 0.1rem solid #3574b6;
}
