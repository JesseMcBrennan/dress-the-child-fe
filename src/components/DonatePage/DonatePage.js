import React, { Component } from 'react';
import { Elements, StripeProvider } from 'react-stripe-elements';
import { withRouter } from 'react-router-dom';

import DonateForm from '../DonateForm/DonateForm';

import './DonatePage.css';

export class DonatePage extends Component {
  render() {
    return (
      <div className="donate-card">
        <StripeProvider apiKey="pk_test_hANH4uSlssQdfIxxHpPwbzPd">
          <div className="donate-section">
            <h1 className="main-title">Donate</h1>
            <hr className="horizontal-rule" />
            <Elements>
              <DonateForm />
            </Elements>
          </div>
        </StripeProvider>
      </div>
    );
  }
}

export default withRouter(DonatePage);
