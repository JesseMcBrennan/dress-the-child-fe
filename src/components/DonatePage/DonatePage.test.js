import React from 'react';
import { shallow } from 'enzyme';
import { DonatePage } from './DonatePage';

describe('DonatePage', () => {
  let mockLocation;
  let wrapper;
  beforeEach(() => {
    mockLocation = {
      pathname: '/donate'
    };
    wrapper = shallow(<DonatePage location={mockLocation} />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
