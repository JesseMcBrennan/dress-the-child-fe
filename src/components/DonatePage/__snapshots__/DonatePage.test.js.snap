// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonatePage should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DonatePage
    location={
      Object {
        "pathname": "/donate",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Provider
        apiKey="pk_test_hANH4uSlssQdfIxxHpPwbzPd"
      >
        <div
          className="donate-section"
        >
          <h1
            className="main-title"
          >
            Donate
          </h1>
          <hr
            className="horizontal-rule"
          />
          <Elements>
            <InjectStripe(DonateForm) />
          </Elements>
        </div>
      </Provider>,
      "className": "donate-card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "apiKey": "pk_test_hANH4uSlssQdfIxxHpPwbzPd",
        "children": <div
          className="donate-section"
        >
          <h1
            className="main-title"
          >
            Donate
          </h1>
          <hr
            className="horizontal-rule"
          />
          <Elements>
            <InjectStripe(DonateForm) />
          </Elements>
        </div>,
        "stripe": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="main-title"
            >
              Donate
            </h1>,
            <hr
              className="horizontal-rule"
            />,
            <Elements>
              <InjectStripe(DonateForm) />
            </Elements>,
          ],
          "className": "donate-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Donate",
              "className": "main-title",
            },
            "ref": null,
            "rendered": "Donate",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "horizontal-rule",
            },
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <InjectStripe(DonateForm) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Provider
          apiKey="pk_test_hANH4uSlssQdfIxxHpPwbzPd"
        >
          <div
            className="donate-section"
          >
            <h1
              className="main-title"
            >
              Donate
            </h1>
            <hr
              className="horizontal-rule"
            />
            <Elements>
              <InjectStripe(DonateForm) />
            </Elements>
          </div>
        </Provider>,
        "className": "donate-card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "apiKey": "pk_test_hANH4uSlssQdfIxxHpPwbzPd",
          "children": <div
            className="donate-section"
          >
            <h1
              className="main-title"
            >
              Donate
            </h1>
            <hr
              className="horizontal-rule"
            />
            <Elements>
              <InjectStripe(DonateForm) />
            </Elements>
          </div>,
          "stripe": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="main-title"
              >
                Donate
              </h1>,
              <hr
                className="horizontal-rule"
              />,
              <Elements>
                <InjectStripe(DonateForm) />
              </Elements>,
            ],
            "className": "donate-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Donate",
                "className": "main-title",
              },
              "ref": null,
              "rendered": "Donate",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "horizontal-rule",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <InjectStripe(DonateForm) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
